packages/typespec-python/src/emitter.ts

```diff
@@ -128,6 +128,12 @@ export async function $onEmit(context: EmitContext<PythonEmitterOptions>) {
         commandArgs.push(`--packaging-files-config='${keyValuePairs.join("|")}'`);
         resolvedOptions["packaging-files-config"] = undefined;
     }
+    if (
+        resolvedOptions["package-pprint-name"] !== undefined &&
+        !resolvedOptions["package-pprint-name"].startsWith('"')
+    ) {
+        resolvedOptions["package-pprint-name"] = `"${resolvedOptions["package-pprint-name"]}"`;
+    }
```

Comment:
The root cause is that [parser.parse_known_args()](https://github.com/Azure/autorest.python/blob/eb05c6c3cee8b3f053a6157e91e63061f37c6efe/packages/autorest.python/autorest/_utils.py#L78) can't handle `name=xx - xx` so we have to make sure it is quoted like `name="xx - xx"`